src/libmongoc/src/mongoc/mongoc-cluster.c

```diff
@@ -3390,17 +3390,15 @@ _mongoc_cluster_run_opmsg_recv (
 
    bson_t body;
 
-   uint32_t op_msg_flags = mcd_rpc_op_msg_get_flag_bits (rpc);
-   cluster->client->in_exhaust = op_msg_flags & MONGOC_OP_MSG_FLAG_MORE_TO_COME;
-
-   if (!mcd_rpc_message_get_body (rpc, &body)) {
+   if ((mcd_rpc_header_get_op_code (rpc) != MONGOC_OP_CODE_MSG) || !mcd_rpc_message_get_body (rpc, &body)) {
```

Comment:
Suggest including the op code in the error message:

```c
   int32_t op_code = mcd_rpc_header_get_op_code (rpc);
   if ((op_code != MONGOC_OP_CODE_MSG) || !mcd_rpc_message_get_body (rpc, &body)) {
      RUN_CMD_ERR (MONGOC_ERROR_PROTOCOL,
                   MONGOC_ERROR_PROTOCOL_INVALID_REPLY,
                   "malformed message from server with op code: %" PRId32,
                   op_code);
      _handle_network_error (cluster, server_stream, error);
      server_stream->stream = NULL;
      network_error_reply (reply, cmd);
      goto done;
   }
```

That may help diagnose if this error is reported again. I am unsure of the underlying cause (server returning unexpected op code?).